impeller/renderer/backend/vulkan/render_pass_vk.cc

```diff
@@ -79,176 +70,65 @@ static std::vector<vk::ClearValue> GetVKClearValues(
-    current_layout = vk::ImageLayout::eGeneral;
-  }
-
-  return CreateAttachmentDescription(desc.format,        //
-                                     desc.sample_count,  //
-                                     load_action,        //
-                                     store_action,       //
-                                     current_layout,
-                                     supports_framebuffer_fetch  //
-  );
-}
-
-static void SetTextureLayout(
-    const Attachment& attachment,
-    const vk::AttachmentDescription& attachment_desc,
-    const std::shared_ptr<CommandBufferVK>& command_buffer,
-    const std::shared_ptr<Texture> Attachment::*texture_ptr) {
-  const auto& texture = attachment.*texture_ptr;
-  if (!texture) {
-    return;
-  }
-  const auto& texture_vk = TextureVK::Cast(*texture);
-
-  if (attachment_desc.initialLayout == vk::ImageLayout::eGeneral) {
-    BarrierVK barrier;
-    barrier.new_layout = vk::ImageLayout::eGeneral;
-    barrier.cmd_buffer = command_buffer->GetEncoder()->GetCommandBuffer();
-    barrier.src_access = vk::AccessFlagBits::eShaderRead;
-    barrier.src_stage = vk::PipelineStageFlagBits::eFragmentShader;
-    barrier.dst_access = vk::AccessFlagBits::eColorAttachmentWrite |
-                         vk::AccessFlagBits::eTransferWrite;
-    barrier.dst_stage = vk::PipelineStageFlagBits::eColorAttachmentOutput |
-                        vk::PipelineStageFlagBits::eTransfer;
-
-    texture_vk.SetLayout(barrier);
-  }
-
-  // Instead of transitioning layouts manually using barriers, we are going to
-  // make the subpass perform our transitions.
-  texture_vk.SetLayoutWithoutEncoding(attachment_desc.finalLayout);
-}
-
 SharedHandleVK<vk::RenderPass> RenderPassVK::CreateVKRenderPass(
     const ContextVK& context,
-    const std::shared_ptr<CommandBufferVK>& command_buffer,
-    bool supports_framebuffer_fetch) const {
-  std::vector<vk::AttachmentDescription> attachments;
-
-  std::vector<vk::AttachmentReference> color_refs;
-  std::vector<vk::AttachmentReference> resolve_refs;
-  vk::AttachmentReference depth_stencil_ref = kUnusedAttachmentReference;
-
-  // Spec says: "Each element of the pColorAttachments array corresponds to an
-  // output location in the shader, i.e. if the shader declares an output
-  // variable decorated with a Location value of X, then it uses the attachment
-  // provided in pColorAttachments[X]. If the attachment member of any element
-  // of pColorAttachments is VK_ATTACHMENT_UNUSED."
-  //
-  // Just initialize all the elements as unused and fill in the valid bind
-  // points in the loop below.
-  color_refs.resize(render_target_.GetMaxColorAttacmentBindIndex() + 1u,
-                    kUnusedAttachmentReference);
-  resolve_refs.resize(render_target_.GetMaxColorAttacmentBindIndex() + 1u,
-                      kUnusedAttachmentReference);
+    const std::shared_ptr<CommandBufferVK>& command_buffer) const {
+  BarrierVK barrier;
```

Comment:
Unconditionally adding barrier SGTM, the old code was making less and less sense.